lms/djangoapps/courseware/management/commands/dump_to_neo4j.py

```diff
@@ -0,0 +1,184 @@
+        relationships = []
+        for item in items:
+            for child_loc in item.get_children():
+                parent_node = location_to_node.get(item.location)
+                child_node = location_to_node.get(child_loc.location)
+                if parent_node is not None and child_node is not None:
+                    relationship = Relationship(parent_node, "PARENT_OF", child_node)
+                    relationships.append(relationship)
+
+        return nodes, relationships
+
+    @staticmethod
+    def coerce_types(value):
+        """
+        Args:
+            value: the value of an xblock's field
+
+        Returns: either the value, a unicode version of the value, or, if the
+        value is iterable, the value with each element being converted to unicode
+        """
+        coerced_value = value
+        if isinstance(value, ITERABLE_NEO4J_TYPES):
+            coerced_value = []
+            for element in value:
+                coerced_value.append(unicode(element))
+            # convert coerced_value back to its original type
+            coerced_value = type(value)(coerced_value)
+
+        # if it's not one of the types that neo4j accepts,
+        # just convert it to unicode
+        elif not isinstance(value, PRIMITIVE_NEO4J_TYPES):
+            coerced_value = unicode(value)
+
+        return coerced_value
+
+
+class Command(BaseCommand):
+    """
+    Command to dump modulestore data to neo4j
+    """
+    @staticmethod
+    def add_to_transaction(neo4j_entities, transaction):
+        """
+        Args:
+            neo4j_entities: a list of Nodes or Relationships
+            transaction: a neo4j transaction
+        """
+        for entity in neo4j_entities:
+            transaction.create(entity)
+
+    def handle(self, *args, **options):  # pylint: disable=unused-argument
+        """
+        Iterates through each course, serializes them into graphs, and saves
+        those graphs to neo4j.
+        """
+        mss = ModuleStoreSerializer()
+        graph = Graph(**settings.NEO4J_CONFIG)
+
+        log.info("deleting existing coursegraph data")
+        graph.delete_all()
+        total_number_of_courses = len(mss.all_courses)
+
+        for index, course in enumerate(mss.all_courses):
+            # first, clear the request cache to prevent memory leaks
+            RequestCache.clear_request_cache()
+
+            log.info(
+                u"Now exporting %s to neo4j: course %d of %d total courses",
+                course.id,
+                index + 1,
+                total_number_of_courses
+            )
+            nodes, relationships = mss.serialize_course(course.id)
+
+            transaction = graph.begin()
+
+            try:
+                self.add_to_transaction(nodes, transaction)
+                self.add_to_transaction(relationships, transaction)
+                transaction.commit()
+
+            except Exception:  # pylint: disable=broad-except
```

Comment:
I've never had a chance to test this on a real data set, just being defensive here